src/thor/costmatrix.cc

```diff
@@ -78,40 +80,49 @@ float CostMatrix::GetCostThreshold(const float max_matrix_distance) {
 // Clear the temporary information generated during time + distance matrix
 // construction.
 void CostMatrix::clear() {
-  // Clear the target edge markings
-  for (auto& iter : *targets_) {
-    iter.second.clear();
-    iter.second.resize(0);
-    iter.second.shrink_to_fit();
+  // resize all relevant structures down to 20 locations
+  if (source_count_ > kMaxLocationCache) {
+    source_edgelabel_.resize(kMaxLocationCache);
+    source_edgelabel_.shrink_to_fit();
+    source_adjacency_.resize(kMaxLocationCache);
+    source_adjacency_.shrink_to_fit();
+    source_edgestatus_.resize(kMaxLocationCache);
+    source_edgestatus_.shrink_to_fit();
+  }
+  if (target_count_ > kMaxLocationCache) {
+    target_edgelabel_.resize(kMaxLocationCache);
+    target_edgelabel_.shrink_to_fit();
+    target_adjacency_.resize(kMaxLocationCache);
+    target_adjacency_.shrink_to_fit();
+    target_edgestatus_.resize(kMaxLocationCache);
+    target_edgestatus_.shrink_to_fit();
   }
-  targets_->clear();
```

Comment:
this is actually what we thought to rectify in https://github.com/valhalla/valhalla/pull/4329#discussion_r1373217995, but I realized it won't work: `targets_` is a robin_hood::unordered_map which didn't implement any `clear()` or anything else. I kept the `targets_.clear()` further down. however, what the old code cleared were the tiny vectors of recorded targets _for every edge_, which could've been a few million and in any case, calling `targets_.clear()` will destruct all those vectors just the line below AFAIK.

Reply:
btw, robin_hood is deprecated! yay..

Reply:
yeah saw that a while ago. the guy made a new project though that is he actively maintaining. honestly though who cares? unlike javascript which is subject ot loads of dependency injection rotting your code over time, a header only c++ library is unlikely to stop working any time soon. im not worried about it in other words :smile:

Reply:
> calling targets_.clear() will destruct all those vectors just the line below AFAIK.

`clear` will simply set the current end pointer of the container back to the beginning AND call the destructor on all the elements which are now past the end pointer but still in the containers allocated space